database/worker.go

```diff
@@ -100,6 +135,17 @@ func (w *Worker) Validate() error {
 		return ErrEmptyWorkerAddress
 	}
 
+	// calculate total size of RunningBuildIds
+	total := 0
+	for _, f := range w.RunningBuildIDs {
+		total += len(f)
+	}
+
+	// verify the RunningBuildIds field is within the database constraints
+	if total > constants.RunningBuildIDsMaxSize {
+		return ErrExceededRunningBuildIDsLimit
+	}
```

Comment:
Since there isn't a defined executor limit for a single worker, and the build id lengths grow over time, I took the approach that if build_id's reach into the millions (up to 9 chara) with commas separating the ids (so 10 chara), this limit accommodates 50 builds.